spec/app/domain/authentication/authn-azure/application_identity_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnAzure::ApplicationIdentity do
+  include_context "azure setup"
+
+  let(:subscription_id_annotation) { double("SubscriptionIdAnnotation") }
+  let(:subscription_id_service_id_scoped_annotation) { double("SubscriptionIdServiceIdAnnotation") }
+
+  let(:resource_group_annotation) { double("ResourceGroupAnnotation") }
+  let(:resource_group_service_id_scoped_annotation) { double("ResourceGroupAnnotation") }
+
+  let(:user_assigned_identity_annotation) { double("UserAssignedIdentityAnnotation") }
+  let(:user_assigned_identity_service_id_scoped_annotation) { double("UserAssignedIdentityAnnotation") }
+
+  let(:non_azure_annotation) { double("NonAzureAnnotation") }
+
+  let(:test_service_id) { "MockService" }
+
+  let(:granular_annotation_type) { "authn-azure/#{test_service_id}" }
```

Comment:
Found anopther one. PLease go over these and verify they are defined in one place only. Do this for all your files.